### ISSUE TITLE
Warn on generic type names in GetInterface

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/ReflectionAnalyzers#23]
Generic interface types may be implemented multiple times.

Refactor to `.FirstOrDefault(interfaceType => interfaceType.IsGenericType && interfaceType.GetGenericTypeDefinition() == ...)`

**Comment by JohanLarsson** [comment]
Do you mean:
```cs
var type = typeof(string).GetInterface("System.Collections.Generic.IEnumerable`1");
```

**Comment by jnm2** [comment]
Yes, but I want a warning even if the type is not known. So this should warn, too:

```c#
public static void Foo(Type type)
{
    type.GetInterface(typeof(IEnumerable<>));
}
```

**Comment by JohanLarsson** [comment]
Yes, reasonable but we can't write a fix, only warn. FirstOrDefault is a big assumption.

**Comment by jnm2** [comment]
@JohanLarsson My thinking was that FirstOrDefault is highly visible and saves typing, but perhaps it should be SingleOrDefault anyway, replacing the ambiguous match exception with an invalid operation exception, and leaving further changes up to the user?

**Comment by JohanLarsson** [comment]
I think we start with a warning notifying about a potential issue. We can always add a fix later.

**Comment by JohanLarsson** [comment]
New analyzer or do we reuse REFL004 for this? Maybe best with a new.
